src/pages/resultsView/pathwayMapper/PathwayMapperTable.tsx

```diff
@@ -0,0 +1,175 @@
+                            </b>
+                        </DefaultTooltip>
+                    </Radio>
+                );
+            },
+            tooltip: <span>Pathway name</span>,
+            filter: (
+                d: IPathwayMapperTable,
+                filterString: string,
+                filterStringUpper: string
+            ) => d.name.toUpperCase().includes(filterStringUpper),
+            sortBy: (d: IPathwayMapperTable) => d.name,
+            download: (d: IPathwayMapperTable) => d.name,
+        };
+
+        this._columns[IPathwayMapperTableColumnType.SCORE] = {
+            name: 'Score',
+            render: (d: IPathwayMapperTable) => (
+                <span>
+                    <b>{d.score.toFixed(2)}</b>
+                </span>
+            ),
+            tooltip: <span>Score</span>,
+            filter: (
+                d: IPathwayMapperTable,
+                filterString: string,
+                filterStringUpper: string
+            ) => (d.score + '').includes(filterStringUpper),
+            sortBy: (d: IPathwayMapperTable) => d.score,
+            download: (d: IPathwayMapperTable) => d.score + '',
+        };
+
+        this._columns[IPathwayMapperTableColumnType.GENES] = {
+            name: 'Genes matched',
+            render: (d: IPathwayMapperTable) => {
+                const geneTextLength = d.genes.join(' ').length;
+
+                return (
+                    <DefaultTooltip
+                        overlay={d.genes.join(' ')}
+                        disabled={geneTextLength < lengthThreshold}
+                    >
+                        <span>
+                            {this.calculateGeneStr(d.genes, lengthThreshold)}
+                        </span>
+                    </DefaultTooltip>
+                );
+            },
+            tooltip: <span>Genes matched</span>,
+            sortBy: (d: IPathwayMapperTable) => d.genes,
+            download: (d: IPathwayMapperTable) => d.genes.toString(),
+        };
+    }
+
+    private dataStore = new SimpleGetterLazyMobXTableApplicationDataStore(
+        () => this.props.data
+    );
+
+    render() {
+        const columns = _.sortBy(this._columns);
+        return (
+            <PathwayMapperTableComponent
+                columns={columns}
+                dataStore={this.dataStore}
+                initialItemsPerPage={10}
+                initialSortColumn={this.props.initialSortColumn}
+                paginationProps={{ itemsPerPageOptions: [10] }}
+                showColumnVisibility={false}
+            />
+        );
+    }
+
+    // It calculates truncated version of gene string without truncating gene names.
+    // That is, instead of truncating a gene name, it removes it all.
+    // For example, if the gene string is MDM2 CDKN2A, instead of truncating it as "MDM2 CKDN ...",
+    // it produces "MDM2 ..."
+    private calculateGeneStr(genesMatched: string[], lengthThreshold: number) {
```

Comment:
this could use unit tests

Reply:
moved the function out of the class and added some unit tests